Zend/tests/dynamic_class_const_fetch.phpt

```diff
@@ -0,0 +1,44 @@
+--TEST--
+Dynamic class constant fetch
+--FILE--
+<?php
+
+class Foo {
+    public const BAR = 'bar';
+}
+
+$bar = 'BAR';
+$ba = 'BA';
+$r = 'R';
+
+var_dump(Foo::{'BAR'});
+var_dump(Foo::{$bar});
+var_dump(Foo::{$ba . $r});
+var_dump(Foo::{strtoupper('bar')});
+
+try {
+    var_dump(Foo::{$barr});
+} catch (Throwable $e) {
+    echo $e->getMessage(), "\n";
+}
+
+$barRef = &$bar;
+var_dump(Foo::{$bar});
+
+try {
+    var_dump(Foo::{strtolower('CLASS')});
+} catch (Throwable $e) {
+    echo $e->getMessage(), "\n";
+}
+
+?>
```

Comment:
`$foo::{$bar}` should be also tested

Reply:
You mean, like `someclass::${'var'}`

Reply:
yes https://3v4l.org/tssrL

and 👍 for this PR